drivers/mspi/mspi_dw.c

```diff
@@ -307,7 +308,9 @@ static void mspi_dw_isr(const struct device *dev)
 		k_sem_give(&dev_data->finished);
 	}
 
+	read_icr(dev);
```

Comment:
Is this actually needed? The flags used for interrupts generation (RXFIS and TXEIS) are cleared by hardware when the FIFOs are read/written, respectively.

> Per the SSI DW documentation, the ICR bit should be read within an
interrupt to clear pending interrupts. Add this to the ISR handler so
interrupts are not fired continuously.

I couldn't find such statement in the documentation that the register should be read. It can be read to clear the interrupts, but I don't think there's an obligation to do so.

Reply:
It appears not to be- I'm guessing I misunderstood the documentation. SPI driver still seems to work without this change locally, so I'll drop it.